core/src/main/java/org/bitcoinj/evolution/QuorumRotationState.java

```diff
@@ -1315,10 +1315,14 @@ public void processDiff(@Nullable Peer peer, QuorumRotationInfo quorumRotationIn
                 log.info("mnList = {} vs qrinfo = {}", mnListTip.getBlockHash(), quorumRotationInfo.getMnListDiffTip().prevBlockHash);
                 log.info("qrinfo {} -> {}", quorumRotationInfo.getMnListDiffTip().prevBlockHash, quorumRotationInfo.getMnListDiffTip().blockHash);
                 log.info("lastRequest: {} -> {}", lastRequest.request.getBaseBlockHashes(), lastRequest.request.getBlockRequestHash());
-                failedAttempts++;
-                log.info("failed attempts {}", failedAttempts);
-                if (failedAttempts > MAX_ATTEMPTS)
+                if (x.requireReset && x.merkleRootMismatch) {
                     resetMNList(true);
+                } else {
+                    incrementFailedAttempts();
+                    log.info("failed attempts {}", getFailedAttempts());
+                    if (reachedMaxFailedAttempts())
+                        resetMNList(true);
+                }
```

Comment:
If merkle roots are mismatched, then we presume that there is a problem the bls key schemes.